### PR TITLE
Revert "Update NuGet global packages cache detection"

### PR DESCRIPTION
**Pull request by premun**
Reverts dotnet/arcade#14763

### CODE REVIEW

**Review by pavel-purma — APPROVED** [review]

**Comment by premun** [comment]
@sharwell we need to urgently flow arcade so I am reverting your changes until you have a fix

File: src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props [hunk 1]
```diff
@@ -5,6 +5,14 @@
      Properties describing the layout of the repo.
   -->
 
+  <PropertyGroup>
+    <!-- Respect environment variable for the NuGet Packages Root if set; otherwise, use the current default location -->
+    <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' != ''">$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)'))</NuGetPackageRoot>
+    <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == '' and '$(NUGET_PACKAGES)' != ''">$([MSBuild]::NormalizeDirectory('$(NUGET_PACKAGES)'))</NuGetPackageRoot>
```
Comment (sharwell) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props:11]: My system will not load any arcade-based code in Visual Studio or build any arcade-based repo from the command line with this invalid code in place.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props [hunk 2]
```diff
@@ -5,6 +5,14 @@
...
+    <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == '' and '$(OS)' == 'Windows_NT'">$([MSBuild]::NormalizeDirectory('$(UserProfile)', '.nuget', 'packages'))</NuGetPackageRoot>
```
Comment (sharwell) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props:12]: My system will not load any arcade-based code in Visual Studio or build any arcade-based repo from the command line with this invalid code in place.

**Comment by sharwell** [comment]
@premun The fix was submitted yesterday in #14868, but I have no good way to validate it

**Comment by premun** [comment]
@sharwell you can possibly:
1. Build your Arcade branch locally, this will produce some `1.0.0-dev` version of the packages somewhere in the `artifacts/packages/` folder.
2. Add the nuget folder to `NuGet.config` of your local `arcade-validation`
3. Set the Arcade versions to `1.0.0-dev`
4. Now `arcade-validation` restores from your local folder
5. Build and run the unit tests locally

**Comment by sharwell** [comment]
@premun Can you clarify how to perform step (3)?

**Comment by premun** [comment]
In `arcade-validation`, in `eng/Versions.props` or wherever (maybe `Directory.Packages.props`), make the repo depend on this local version.

**Pull request merged**